modules/ROOT/pages/dataweave-troubleshoot.adoc

```diff
@@ -109,3 +115,74 @@ Trace:
+----
+org.mule.runtime.api.el.ExpressionExecutionException: Stack Overflow. Max stack is 256
+----
+
+You can configure the maximum stack size by using the property `com.mulesoft.dw.stacksize`.
+
+== Output Mismatch When Undefined
+
+Unlike transformations, DataWeave expressions do not require you to define an output format because DataWeave can infer the output based on the expression and the variables you use. Occasionally, the inference process results in a mismatch between the inferred type and the expected type. To resolve this issue, you must make the output explicit. Common examples of this situation occur when:
+
+* <<xml_extraction>>
+* <<multipart_handling>>
+* <<manipulating_text>>
+
+[[xml_extraction]]
+=== Extracting Data from XML
+
+When extracting a String, for example, from an XML payload with the expression `payload.order.product.model`, DataWeave  infers an XML output based on the payload format. In such cases, an error similar to the following one occurs:
+
+[source,weave, linenums]
+----
+"Trying to output non-whitespace characters outside main element tree (in prolog or epilog), while writing Xml at ." evaluating expression: "payload.order.product.model".
+----
+
+For such an error, you must make the output format explicit, for example: `output text/plain --- payload.order.product.model`.
+
+[[multipart_handling]]
+=== Handling Multipart Entries
+
+A common inference error occurs with multipart data, which has a very specific structure. Consider a multipart payload and the expression `dw::core::Objects::keySet(payload.parts)`. Without an explicit output format, DataWeave must infer, based on the payload type, that you intend to output multipart content. In this case, an error similar to the following is thrown:
+
+[source,weave, linenums]
+----
+"Expecting type is {
+  preamble?: String,
+  parts: {
+    _*: {
+      headers: Object,
+      content: Any
+    }
+  }
+} but got Array, while writing MultiPart.
+Trace:
+  at main (Unknown)" evaluating expression: "dw::core::Objects::keySet(payload.parts)".
+----
+
+To resolve this issue, you must define an output format, for example: `output application/json --- dw::core::Objects::keySet(payload.parts)`
+
+[[manipulating_text]]
+=== Manipulating Text Data
+
+Text data can be used to create a more complex object. However, if you do not define the output format for the input text data, DataWeave will infer that it needs to use the plain text writer for the output. The expression `payload splitBy ' '`, for example, will fail with an error similar to:
+
+[source,weave, linenums]
+----
+"Text plain writer is unable to write Array.
+Reason:
+Cannot coerce Array (org.mule.weave.v2.model.values.ArrayValue$ArraySeqArrayValue@21bec687) to String
+Trace:
+  at main (Unknown), while writing TextPlain.
+Trace:
+  at main (Unknown)" evaluating expression: "payload splitBy ' '".
+----
+
+Making the output explicit solves the issue: `output application/java --- payload splitBy ''`
```

Comment:
Is there another part to this sentence, or should this end with a period?
The number of backticks/quotation marks does not match the example above it.

Reply:
That's just what the error looks like in the console.